stdlib/zlib.pyi

```diff
@@ -27,18 +28,28 @@ Z_TREES: Final = 6
 
 class error(Exception): ...
 
+# This class is not exposed at runtime. It calls itself zlib.Compress.
+@final
+@type_check_only
 class _Compress:
-    def compress(self, data: ReadableBuffer) -> bytes: ...
-    def flush(self, mode: int = ...) -> bytes: ...
-    def copy(self) -> _Compress: ...
+    def __copy__(self, /) -> Self: ...
+    def __deepcopy__(self, memo: Any, /) -> Self: ...
+    def compress(self, data: ReadableBuffer, /) -> bytes: ...
+    def flush(self, mode: int = 4, /) -> bytes: ...
+    def copy(self, /) -> _Compress: ...
```

Comment:
As Jelle pointed out to me in https://github.com/python/typeshed/pull/13013#discussion_r1844343209, we don't usually mark `self` as positional-only; we can reduce the diff here a little bit by avoiding those changes:

```suggestion
    def __copy__(self) -> Self: ...
    def __deepcopy__(self, memo: Any, /) -> Self: ...
    def compress(self, data: ReadableBuffer, /) -> bytes: ...
    def flush(self, mode: int = 4, /) -> bytes: ...
    def copy(self) -> _Compress: ...
```